### PR TITLE
Add section enabled rules from previous sections in when rules

### PR DESCRIPTION
**Pull request by petechd**
### What is the context of this PR?
These code changes are added to ensure implement section enabled when rule evaluation, including the ability to reference answer IDs from a previous section, questionnaire is updated correctly to reflect the true state of the dependent questions in other sections. Scenarios are captured in new functional tests. To understand the changes follow new test scenarios.

### How to review 
Run functional tests, check new code, possibly look at previous PR which did the groundwork -> https://github.com/ONSdigital/eq-questionnaire-runner/pull/809

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by petechd** [comment]
[Section enabled routing path in when rules (3 SP)](https://trello.com/c/byKaYwZh/5144-section-enabled-routing-path-in-when-rules-3-sp)

**Review by berroar — APPROVED** [review]
Looks good and working as described

File: tests/functional/spec/routing_and_skipping_section_dependencies.spec.js [hunk 1]
```diff
@@ -122,6 +124,28 @@ describe("Routing and skipping section dependencies", () => {
     });
   });
 
+  describe("Given the routing and skipping section dependencies questionnaire", () => {
+    it("When I answer 'No' to skipping the section question and 'Yes' to enable section question, Then the household summary will be visible on the hub", () => {
```
Comment (rmccar) [inline tests/functional/spec/routing_and_skipping_section_dependencies.spec.js:128]:
```suggestion
    it("When I answer 'No' to skipping the section question and 'Yes' to enable the section question, Then the household summary will be visible on the hub", () => {
```

File: tests/functional/spec/routing_and_skipping_section_dependencies.spec.js [hunk 2]
```diff
@@ -122,6 +124,28 @@ describe("Routing and skipping section dependencies", () => {
...
+      answerNoToSkipEnableQuestionAndYesToEnableSection();
+
+      expect($(HubPage.summaryRowLink("household-section")).isExisting()).to.be.true;
+    });
+    it("When I answer 'No' to skipping the section question and 'No' to enable section question, Then the household summary will not be visible on the hub", () => {
```
Comment (rmccar) [inline tests/functional/spec/routing_and_skipping_section_dependencies.spec.js:133]:
```suggestion
    it("When I answer 'No' to skipping the section question and 'No' to enable the section question, Then the household summary will not be visible on the hub", () => {
```

File: tests/functional/spec/routing_and_skipping_section_dependencies.spec.js [hunk 3]
```diff
@@ -122,6 +124,28 @@ describe("Routing and skipping section dependencies", () => {
...
+      answerNoToSkipEnableQuestionAndNoToEnableSection();
+
+      expect($(HubPage.summaryRowLink("household-section")).isExisting()).to.be.false;
+    });
+  });
+
+  describe("Given the routing and skipping section dependencies questionnaire and I answered 'No' to skipping the section question and 'Yes' to enable section question", () => {
```
Comment (rmccar) [inline tests/functional/spec/routing_and_skipping_section_dependencies.spec.js:140]:
```suggestion
  describe("Given the routing and skipping section dependencies questionnaire and I answered 'No' to skipping the section question and 'Yes' to enable the section question", () => {
```

File: tests/functional/spec/routing_and_skipping_section_dependencies.spec.js [hunk 4]
```diff
@@ -122,6 +124,28 @@ describe("Routing and skipping section dependencies", () => {
...
+    it("When I change my answer to skipping section question to 'No', Then the household summary will not be visible on the hub", () => {
```
Comment (rmccar) [inline tests/functional/spec/routing_and_skipping_section_dependencies.spec.js:141]:
```suggestion
    it("When I change my answer to skipping the section question to 'No', Then the household summary will not be visible on the hub", () => {
```

**Review by rmccar — APPROVED** [review]
Works as expected, just a few typos

File: app/questionnaire/router.py [hunk 5]
```diff
@@ -343,6 +343,11 @@ def _is_section_enabled(self, section: Mapping) -> bool:
             return True
 
         enabled = section["enabled"]
+
+        routing_path_block_ids = self._path_finder.get_when_rules_block_dependencies(
+            section["id"]
+        )
```
Comment (MebinAbraham) [inline app/questionnaire/router.py:349]: We should pass this into the old engine as well. Not need to test.

File: schemas/test/en/test_new_routing_and_skipping_section_dependencies.json [hunk 6]
```diff
@@ -58,6 +58,77 @@
...
+                                            },
+                                            {
+                                                "label": "No",
+                                                "value": "No"
+                                            }
+                                        ],
+                                        "type": "Radio",
+                                        "validation": {
+                                            "messages": {}
+                                        }
```
Comment (MebinAbraham) [inline schemas/test/en/test_new_routing_and_skipping_section_dependencies.json:117]: Can be removed.

**Review by MebinAbraham — COMMENTED** [review]
LGTM 👍

File: schemas/test/en/test_new_routing_and_skipping_section_dependencies.json [hunk 7]
```diff
@@ -58,6 +58,77 @@
...
```
[the remark this answers is not in the document]
Comment (petechd) [reply schemas/test/en/test_new_routing_and_skipping_section_dependencies.json]: Removed.

File: schemas/test/en/test_new_routing_and_skipping_section_dependencies.json (hunk 6 above)
Comment (petechd) [reply schemas/test/en/test_new_routing_and_skipping_section_dependencies.json]: Removed.

File: app/questionnaire/router.py (hunk 5 above)
Comment (petechd) [reply app/questionnaire/router.py:349]: Ok added and also `current_location=None` since we do that in the new evaluator as well (for clarity?).

**Review by MebinAbraham — APPROVED** [review]

**Pull request merged**